doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1183,6 +1183,8 @@ Here are the syntax elements used by the various notation commands.
       | in custom @ident {? at level @natural }
       | {+, @ident } at @level
       | @ident at @level {? @binder_interp }
+      | {+, @ident } in scope @scope_name
+      | @ident in scope @scope_name
```

Comment:
These productions should be updated by running `make doc_gram_rsts` rather than by hand.  Please run that command and include `fullGrammar` and `orderedGrammar` in the PR.  You'll need to remove `| only_parsing` from `common.edit_mlg`.

Also, BTW, the second added production is not needed--the first added production is a superset of it.

Reply:
I get this
```
Error: Can't find 'IDENT; "," LIST1 IDENT SEP "," "at" level' in edit for 'syntax_modifier'
1 error(s)
```
and
```shell
$ git diff --stat doc/tools/docgram/fullGrammar | cat
 doc/tools/docgram/fullGrammar | 2120 ++++++++++++++++++++---------------------
 1 file changed, 1058 insertions(+), 1062 deletions(-)
```
So I guess something went wrong. I can include it, bit the diff is just too much.

Reply:
Note that I had to factor two rules to make camlp5 happy:
https://github.com/coq/coq/pull/13965/files#diff-7a77660cab5c23a4ec78a2c6726d6808e90d7dfb364098ece99d13920e23648bR1227

Reply:
> So I guess something went wrong

Not really.  It's just that `common.edit_mlg` tries to match and edit that production, which you just changed.  So the edit instruction needs to be updated.

> Note that I had to factor two rules to make camlp5 happy:

Sure.  We recombine them for the doc in `common.edit_mlg` since that's an uninteresting implementation detail.

Change `common.edit_mlg` so it has:

```
syntax_modifier: [
| DELETE "in" "custom" IDENT
| REPLACE "in" "custom" IDENT; "at" "level" natural
| WITH "in" "custom" IDENT OPT ( "at" "level" natural )
| DELETE IDENT; "in" "scope" IDENT
| REPLACE IDENT; "," LIST1 IDENT SEP "," [ "at" level | "in" "scope" IDENT ]
| WITH LIST1 IDENT SEP "," [ "at" level | "in" "scope" IDENT ]
]
```

And `orderedGrammar` so it has:

syntax_modifier: [
| "at" "level" natural
| "in" "custom" ident OPT ( "at" "level" natural )
| LIST1 ident SEP "," [ "at" level | "in" "scope" ident ]
| ident "at" level OPT binder_interp
| ident explicit_subentry
| ident binder_interp
| "left" "associativity"
| "right" "associativity"
| "no" "associativity"
| "only" "parsing"
| "only" "printing"
| "format" string OPT string
]
```

Then rerun `make doc_gram_rsts` to update the rsts.

I tried to push an update on your branch, but it said permission denied even for a `push -f`.  It may be I didn't have the right command.

Reply:
That is weird, anyway I could update the grammar files you suggested and not it worked.
I've pushed also the ful grammar which has a large diff, I don't know why.

Reply:
After rebasing on master the diff on full grammar is reasonable again (I got a conflict and I re-run the tool).